interactive_engine/executor/ir/core/src/glogue/pattern.rs

```diff
@@ -820,6 +872,77 @@ fn has_expr_eq(expr: &common_pb::Expression) -> bool {
     false
 }
 
+/// Pick up EdgeExpand Operator from a binder in pb::Pattern's sentences
+fn get_edge_expand_from_binder<'a, 'b>(
+    binder: &'a pb::pattern::Binder, edge_id: PatternId,
+    edge_data_map: &'b mut BTreeMap<PatternId, PbEdgeOrPath>,
+) -> IrPatternResult<&'a pb::EdgeExpand> {
+    use pb::pattern::binder::Item as BinderItem;
+    if let Some(BinderItem::Path(path_expand)) = binder.item.as_ref() {
+        let hop_range = path_expand
+            .hop_range
+            .as_ref()
+            .ok_or(ParsePbError::EmptyFieldError("pb::PathExpand::hop_range".to_string()))?;
+        if hop_range.lower < 1 {
+            // The path with range from 0 cannot be translated to oprs that can be intersected.
+            return Err(IrPatternError::Unsupported(format!(
+                "PathExpand in Pattern with lower range of {:?}",
+                hop_range.lower
+            )))?;
+        }
+        let expand_base = path_expand
+            .base
+            .as_ref()
+            .ok_or(ParsePbError::EmptyFieldError("PathExpand::base in Pattern".to_string()))?;
+        expand_base
+            .get_v
+            .as_ref()
+            .ok_or(IrPatternError::Unsupported(
+                "PathExpand::base in Pattern doesn't have GetV".to_string(),
+            ))?;
+        let edge_expand = expand_base
+            .edge_expand
+            .as_ref()
+            .ok_or(ParsePbError::EmptyFieldError("PathExpand::base::EdgeExpand in Pattern".to_string()))?;
+        edge_data_map.insert(edge_id, PbEdgeOrPath::from(path_expand.clone()));
+        Ok(edge_expand)
+    } else if let Some(BinderItem::Edge(edge_expand)) = binder.item.as_ref() {
+        if edge_expand.expand_opt == pb::edge_expand::ExpandOpt::Degree as i32 {
+            Err(IrPatternError::Unsupported(
+                "Expand Degree hasn't been supported in pattern match".to_string(),
+            ))?
+        }
+        edge_data_map.insert(edge_id, PbEdgeOrPath::from(edge_expand.clone()));
+        Ok(edge_expand)
+    } else {
+        Err(IrPatternError::InvalidExtendPattern(
+            "Only can pick up `EdgeExpand` from BinderItem::Path and BinderItem::Edge".to_string(),
+        ))?
+    }
+}
+
+/// Pick up pb::QueryParams for vertex from a binder in pb::Pattern's sentences
+fn get_vertex_parameters_from_binder(binder: &pb::pattern::Binder) -> Option<pb::QueryParams> {
+    use pb::pattern::binder::Item as BinderItem;
+    if let Some(BinderItem::Path(path_expand)) = binder.item.as_ref() {
+        path_expand
+            .base
+            .as_ref()
+            .and_then(|expand_base| expand_base.get_v.as_ref())
+            .and_then(|get_v| get_v.params.as_ref())
+            .cloned()
```

Comment:
No need to get_vertex_parameters from `PathExpand`, as all params in `PathExpand` has already be preserved in the Pattern Edge. 
Actually, `PathExpand` may have two different kinds of vertex filters:
1. GetV with filter in ExpandBase of path, which specifies the conditions for EVERY VERTEX in path, and as said, this info has already preserved in  the edge map.
2. EndV with filter followed by `PathExpand`, which specifies the condition for the LAST VERTEX in path. And this is the one we need to preserve here.